contracts/finance/andromeda-set-amount-splitter/src/contract.rs

```diff
@@ -0,0 +1,315 @@
+    amp::messages::AMPPkt,
+    common::{actions::call_action, encode_binary, Milliseconds, MillisecondsDuration},
+    error::ContractError,
+};
+use andromeda_std::{ado_contract::ADOContract, common::context::ExecuteContext};
+use cosmwasm_std::{
+    attr, coins, ensure, entry_point, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, Env,
+    MessageInfo, Reply, Response, StdError, SubMsg,
+};
+use cw_utils::nonpayable;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-set-amount-splitter";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+// 1 day in seconds
+const ONE_DAY: u64 = 86_400;
+// 1 year in seconds
+const ONE_YEAR: u64 = 31_536_000;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let splitter = match msg.lock_time {
+        Some(ref lock_time) => {
+            // New lock time can't be too short
+            ensure!(
+                lock_time.get_time(&env.block).seconds() >= ONE_DAY,
+                ContractError::LockTimeTooShort {}
+            );
+
+            // New lock time can't be too long
+            ensure!(
+                lock_time.get_time(&env.block).seconds() <= ONE_YEAR,
+                ContractError::LockTimeTooLong {}
+            );
+            Splitter {
+                recipients: msg.recipients.clone(),
+                lock: lock_time.get_time(&env.block),
+            }
+        }
+        None => {
+            Splitter {
+                recipients: msg.recipients.clone(),
+                // If locking isn't desired upon instantiation, it's automatically set to 0
+                lock: Milliseconds::default(),
+            }
+        }
+    };
+    // Save kernel address after validating it
+
+    SPLITTER.save(deps.storage, &splitter)?;
+
+    let inst_resp = ADOContract::default().instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        &deps.querier,
+        info,
+        BaseInstantiateMsg {
+            ado_type: CONTRACT_NAME.to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            kernel_address: msg.kernel_address.clone(),
+            owner: msg.owner.clone(),
+        },
+    )?;
+
+    msg.validate(deps.as_ref())?;
+
+    Ok(inst_resp)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(_deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    if msg.result.is_err() {
+        return Err(ContractError::Std(StdError::generic_err(
+            msg.result.unwrap_err(),
+        )));
+    }
+
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let ctx = ExecuteContext::new(deps, info, env);
+
+    match msg {
```

Comment:
No need to use `call_action` here?

Reply:
Just went through some of our contracts and all of them place `call_action` in `handle_execute`, not `execute`.